backends/cortex_m/ops/op_quantized_linear.cpp

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include "cortex_m_ops_common.h"
+
+// Include CMSIS-NN headers with C linkage
+extern "C" {
+#include "arm_nnfunctions.h"
+}
+
+namespace cortex_m {
+namespace native {
+using KernelRuntimeContext = torch::executor::KernelRuntimeContext;
+
+// Pass allocates a flat scratch buffer:
+// [------------------- scratch_buffer -----------------------]
+// |<- CMSIS-NN workspace ->|<--- kernel_sum_state struct --->|
+//
+// Buffer pointers:
+// ^                        ^                                 ^
+// scratch_ptr(start)   scratch_ptr + cmsis_scratch   scratch_ptr + total_size
+//
+// - CMSIS-NN workspace: used by CMSIS-NN kernels for temporary data
+// - Always give CMSIS-NN the start of the buffer for alignment
```

Comment:
can we have a fat-header i.e. 32B to solve alignment problem? The reason I Don't like trailing header is because OOB read (if not also write) to match SIMD width is an acceptable practice.